Cython/Compiler/Parsing.py

```diff
@@ -2899,7 +2899,8 @@ def p_c_func_declarator(s, pos, ctx, base, cmethod_flag):
     ellipsis = p_optional_ellipsis(s)
     s.expect(')')
     nogil = p_nogil(s)
-    exc_val, exc_check = p_exception_value_clause(s)
+    exc_val, exc_check = p_exception_value_clause(s, ctx)
+    nogil = nogil or p_nogil(s)
```

Comment:
@scoder re: your comment [here](https://github.com/cython/cython/pull/4670#discussion_r854489596), I believe this will allow placing `nogil` either before or after the exception value -- was this what you were looking for?

If we want to enable this only for Cython < 3.0, would parsing `Cython.__version__` be the right way to go about that? (I couldn't find similar examples in the codebase).

Reply:
In terms of Cython <3.0 you'd make a separate PR against the 0.29.x branch with just this change and whatever tests were needed.

I wasn't sure from the comment when it'd be deprecated, but I'd imagine it'd probably have to be allowed in Cython 3 initially. I wouldn't parse `Cython.__version__`, just commit different code to the two branches

Reply:
@da-woods given your changes in https://github.com/cython/cython/pull/4903, do we now want to remove this line?

Reply:
I wasn't sure. I think it might be worth warning for the non-preferred version in Cython 3. I don't think we should ban force one version right away.

Reply:
Yeah, it's usually better to warn first. This isn't a required incompatible change, it's just something we should get straight eventually and push users into doing "right". I described in https://github.com/cython/cython/pull/4670#discussion_r854489596, which you already referenced above, what we should allow.

Reply:
@scoder I just added a warning - could you please take a look at the wording?

Reply:
Reads well to me.